src/ResearchMethod.java

```diff
@@ -62,4 +62,29 @@ public void printDevices(List<Product> products) {
                     "\n \n");
         }
     }
+    public void searchBySalePrice(float minPrice, float maxPrice) {
```

Comment:
Sinceramente ho dubbi su cosa la consegna richieda nello specifico. C'è già un metodo che ricerca per range di prezzo nei task. Questo quindi immagino non dovrebbe ricercare per range ma proprio per salePrice. Cosa voglia dire però non lo so

Reply:
cambiato il nome del metodo come stabilito in call. Se è tutto ok attendo approvazione e merge